tests/integration/HACKING.md

```diff
@@ -0,0 +1,468 @@
+# Integration Testing with Girder Worker
+TODO
+
+# Architectural notes for Developers (and Masochists)
+
+The following sections contain information about the implementation details of Girder Worker. They are not meant for general consumption,  but may be useful in explaining some of the design decisions that were made for developers who wish to extend or contribute to Girder Worker.
+
+## The 'core' plugin
+TODO
+
+## Celery Hooks and the reserved keywords
+TODO
+
+## The jobInfoSpec and the task job\_manager
+TODO
+
+
+## Issues with the Traditional Method
+The fundamental issue with the Traditional Method is how does one synchronize Celery's configuration across producer (Girder) and consumer (Girder Worker) processes, keeping in mind that these processes may be taking place on two (or more) different physical machines. Traditionally Celery solves this problem through the Python import mechanism. The Celery application is configured as apart of the python package in which it is defined. By importing the application you get the same configuration on the producer as on the consumer. Assuming you are using the same version on both producer and consumer, then application is imported consumer-side when running ```girder-worker``` and is imported implicitly producer-side when importing a Celery Task object (e.g. ```from my_custom_package.tasks import fibonacci```).
+
+The traditional method does not use Celery Task objects to produce messages (e.g. with the ```.delay(...)``` or ```.apply_async(...)``` task methods). Instead it uses the ```Celery.send_task(...)``` method which is set on the Celery application object. ```.send_taks(...)``` is the lowest level API for producing messages in Celery.  Rather than using a task object which is generated from a wrapped Python function it uses the task "name" (usually the string-ified importable path to the task). The advantage here is that the task does not need to be importable on the producer-side system, it doesn't even have to be installed at all. However,  because the task object itself is not producing the message all producer-side task level configuration is lost when using ```Celery.send_task(...)```. This might not be a deal-breaker except that the application object on the Girder producer-side is a *different application object* than on the Girder Worker consumer-side. The worker plugin does not import the ```girder_worker.app``` application object, instead it fabricates a new ```Celery(...)``` object before calling ```.send_task(...)```. While on the one hand this means the girder_worker package doesn't need to be installed in the girder environment it *also* means that the Celery application object on the producer-side is a differently configured object then on the consumer-side. Maybe that isn't a problem for you and your use-case (Celery has excellent defaults), but if you need more sophisticated configurations then currently the "Celery" method for remote task execution is your only option.
```

Comment:
it uses the celery.send_task method which is set on the celery application object.  what does `set` mean here?

s/send_taks/send_task/

Maybe that isn't a problem ... can you give an example or two of why this would be a problem?